### PR TITLE
Refactor net-vpc module for Terraform 1.3

### PR DESCRIPTION
**Pull request by ludoo**
Currently in draft since all blueprints and FAST stages using this module still need to be updated.

### CODE REVIEW

**Review by juliocc — APPROVED** [review]
❤️

File: blueprints/data-solutions/cloudsql-multiregion/main.tf [hunk 1]
```diff
@@ -36,7 +36,7 @@ locals {
       local.data_eng_principals_iam,
       [module.service-account-sql.iam_email]
     )
-    # compute engeneering 
+    # compute engeneering
```
[the remark this answers is not in the document]
Comment (ludoo) [reply blueprints/data-solutions/cloudsql-multiregion/main.tf:39]: nice find :)

File: modules/net-vpc/variables.tf [hunk 2]
```diff
@@ -41,44 +41,19 @@ variable "description" {
...
-}
-
-variable "log_configs" {
-  description = "Map keyed by subnet 'region/name' of optional configurations for flow logs when enabled."
-  type        = map(map(string))
-  default     = {}
-}
-
 variable "mtu" {
   description = "Maximum Transmission Unit in bytes. The minimum value for this field is 1460 and the maximum value is 1500 bytes."
```
Comment (sruffilli) [inline modules/net-vpc/variables.tf:55]: Nit: I'd mention that default is 1460.

File: modules/net-vpc/variables.tf [hunk 3]
```diff
@@ -154,42 +130,45 @@ variable "shared_vpc_service_projects" {
...
+    ipv6 = optional(object({
+      access_type           = optional(string)
+      enable_private_access = optional(bool, true)
+    }))
+    secondary_ip_ranges = optional(map(string))
   }))
   default = []
 }
 
 variable "subnets_proxy_only" {
```
[the remark this answers is not in the document]
Comment (ludoo) [reply modules/net-vpc/variables.tf:164]: we discussed it yesterday and it makes everything ugly and more complex, we also have separate resources which we plan on keeping

**Pull request merged**